client/getResource_test.go

```diff
@@ -59,18 +58,24 @@ func TestClientGetResource(t *testing.T) {
 		},
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), TestTimeout)
+	c, err := NewTestSecureClient()
+	require.NoError(t, err)
+	defer func() {
+		err := c.Close(context.Background())
+		require.NoError(t, err)
+	}()
+	ctx, cancel := context.WithTimeout(context.Background(), TestTimeout*8)
 	defer cancel()
-
-	c := NewTestClient()
-	defer c.Close(context.Background())
-
+	deviceID, err = c.OwnDevice(ctx, deviceID)
+	require.NoError(t, err)
+	defer func() {
+		err := c.DisownDevice(ctx, deviceID)
+		require.NoError(t, err)
+	}()
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			ctx, cancel := context.WithTimeout(ctx, time.Second)
-			defer cancel()
 			var got map[string]interface{}
-			err := c.GetResource(ctx, tt.args.deviceID, tt.args.href, &got, tt.args.opts...)
```

Comment:
I think deviceID can be removed from args, since it is no longer used.

Reply:
I want to test it with non exist device so will leave it and use proper `tt.args.deviceID`